drivers/analog-io/max2201x/iio_max2201x.c

```diff
@@ -0,0 +1,919 @@
+				  const struct iio_ch_info *channel,
+				  intptr_t priv);
+
+static int max2201x_iio_write_offset(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv);
+
+static int max2201x_iio_write_gain(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv);
+
+static int max2201x_iio_write_slew_rate(void *dev, char *buf, uint32_t len,
+					const struct iio_ch_info *channel,
+					intptr_t priv);
+
+static int max2201x_iio_read_step_size_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_read_update_rate_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_read_range_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_read_enable_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_write_op_mode(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel,
+				      intptr_t priv);
+
+static int max2201x_iio_read_op_mode_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_write_ao_config(void *dev, char *buf, uint32_t len,
+					const struct iio_ch_info *channel,
+					intptr_t priv);
+
+static int max2201x_iio_read_polarity_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_read_cm_sense_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_write_config(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv);
+
+static int max2201x_iio_read_config_available(void *dev, char *buf,
+		uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv);
+
+static int max2201x_iio_write_timeout(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel,
+				      intptr_t priv);
+
+static int max2201x_iio_read_timeout_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+
+static int max2201x_iio_read_interrupt(void *dev, char *buf, uint32_t len,
+				       const struct iio_ch_info *channel,
+				       intptr_t priv);
+
+static int max2201x_iio_write_int_en(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv);
+
+static int max2201x_iio_reg_read(struct max2201x_iio_desc *iio_desc,
+				 uint32_t reg, uint32_t *readval);
+
+static int max2201x_iio_reg_write(struct max2201x_iio_desc *iio_desc,
+				  uint32_t reg, uint32_t writeval);
+
+static struct iio_attribute max2201x_attrs[] = {
```

Comment:
regarding attributes, the names should be compatible with attributes name from Linux IIO sub-system, for compatibility reasons, so please take a look at this list: https://github.com/torvalds/linux/blob/master/drivers/iio/industrialio-core.c#L155 and only leave out the attributes available there. the other attributes should be moved in max2201x_debug_attrs